inst/table.m

```diff
@@ -424,19 +424,26 @@ function prettyprint (this)
     ## -*- texinfo -*-
     ## @node table.size
     ## @deftypefn {Method} {@var{sz} =} size (@var{obj})
+    ## @deftypefnx {Method} {[@var{nr}, @var{nv}] =} size (@var{obj})
     ##
     ## Gets the size of a table.
     ##
     ## For tables, the size is [number-of-rows x number-of-variables].
     ## This is the same as @code{[height(obj), width(obj)]}.
     ##
     ## @end deftypefn
-    function out = size (this, dim)
-      if nargin == 1
-        out = [height(this), width(this)];
+    function [out, nv] = size (this, dim)
+      if nargout == 1
```

Comment:
So, here's the deal: instead of just handling the `nargout ==1` and `nargout == 2` cases, I think we need to handle _any_ number of argouts.

Zero argouts should have the same behavior as `nargout == 1`. The `if nargout == 1 ... else` logic here is going to toss `nargout == 0` into the `nargout == 2` bin and give you just the height, I think. That's wrong; you should get the full 2-D size in that case.

And we need to extend this to handle nargout > 2: in Octave, arrays are actually many- or infinite-dimensioned, and a 2-D array like a table array actually has an infinite number of trailing singleton dimensions whose size is all 1. So if I do this:

```octave
x = [1; 2; 3];
y = [420; 69; 42];
tbl = table(x, y);
[d1, d2, d3, d4, d5, d6] = size(tbl)
```

I think that should actually work, and you should get:
* d1 = 3
* d2 = 2
* d3, d4, d5, d6 all = 1

Basically, I think we need to extend this logic to take any number of argouts, and when nargout > 2, fill all the nargout{i>2} with 1s.

And in the 2-argin case, whenever dim is > 2, return 1.

What do you think?

Reply:
Ok, I agree that the correct behavior for a nargout > 2 is 1's for the additional dimensions (not an error as we have now). I'll try to fix it tomorrow.

Reply:
Cool beans. No rush – it's the weekend, and this is a volunteer project. Get to it when you get to it. I'll wait on your PR revision before making any other moves here.